Sources/Core/View/UIKit/SnackbarUIView.swift

```diff
@@ -0,0 +1,380 @@
+        self.horizontalStackView.alignment = .center
+        self.horizontalStackView.axis = .horizontal
+    }
+
+    private func setupLabel() {
+        self.label.textColor = self.viewModel.foregroundColor.uiColor
+        self.label.font = self.viewModel.textFont.uiFont
+        self.label.adjustsFontForContentSizeCategory = true
+        self.label.setContentCompressionResistancePriority(.required, for: .vertical)
+        self.label.translatesAutoresizingMaskIntoConstraints = false
+        self.labelContainer.addSubview(self.label)
+
+        self.labelTrailingAnchorConstraint = self.label.trailingAnchor.constraint(equalTo: self.labelContainer.trailingAnchor)
+        self.labelTrailingAnchorConstraint.priority = .required - 1
+
+        NSLayoutConstraint.activate([
+            self.label.leadingAnchor.constraint(equalTo: self.labelContainer.leadingAnchor),
+            self.label.topAnchor.constraint(greaterThanOrEqualTo: self.labelContainer.topAnchor),
+            self.label.centerYAnchor.constraint(equalTo: self.labelContainer.centerYAnchor),
+            self.labelTrailingAnchorConstraint
+        ])
+    }
+
+    // MARK: - Setup Layout
+    private func setupVerticalStackViewLayout() {
+        NSLayoutConstraint.activate([
+            self.verticalStackView.leadingAnchor.constraint(equalTo: self.leadingAnchor, constant: self.theme.layout.spacing.large),
+            self.verticalStackView.trailingAnchor.constraint(equalTo: self.trailingAnchor, constant: -self.theme.layout.spacing.medium),
+            self.verticalStackView.topAnchor.constraint(equalTo: self.topAnchor, constant: self.theme.layout.spacing.medium),
+            self.verticalStackView.centerYAnchor.constraint(equalTo: self.centerYAnchor),
+        ])
+    }
+
+    private func setupAdvancedLayout() {
+        // Image
+        if let imageView {
+            self.addMinimumTopSpacing(self.theme.layout.spacing.small, to: imageView)
+        }
+
+        // Label
+        self.addMinimumTopSpacing(self.theme.layout.spacing.large, to: self.label)
+        self.updateLabelTrailingConstraint()
+
+        // Button
+        if let buttonView {
+            self.addMinimumTopSpacing(self.theme.layout.spacing.small, to: buttonView)
+        }
+    }
+
+    private func addMinimumTopSpacing(_ constant: CGFloat, to view: UIView) {
+        let topAnchorConstraint = view.topAnchor.constraint(
+            greaterThanOrEqualTo: self.topAnchor,
+            constant: constant
+        )
+        topAnchorConstraint.priority = .required - 1
+        topAnchorConstraint.isActive = true
+    }
+
+    /// Label trailing space management.
+    /// Spacing will be `0` if `button` exists and is in the `horizontalStackView`.
+    /// Spacing will be `(spacing.large - spacing.medium)` otherwise to be further away from the borders.
+    private func updateLabelTrailingConstraint() {
+        var trailingConstant = -(self.theme.layout.spacing.large - self.theme.layout.spacing.medium)
```

Comment:
can be managed on the Viewmodel ..?

Reply:
Can't be done because UIKit and SwiftUI have different behaviours regarding the spacings